misk/src/main/kotlin/misk/moshi/wire/WireMessageAdapter.kt

```diff
@@ -0,0 +1,90 @@
+package misk.moshi.wire
+
+import com.google.common.base.CaseFormat
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.JsonReader
+import com.squareup.moshi.JsonWriter
+import com.squareup.moshi.Moshi
+import com.squareup.wire.Message
+import com.squareup.wire.WireField
+import java.lang.reflect.Type
+
+/** Json marshaling for Wire messages, correctly using Builders to construct properly formed type */
+internal class WireMessageAdapter(
+        messageType: Class<Message<*, *>>,
+        private val moshi: Moshi,
+        private val jsonCaseFormat: CaseFormat = CaseFormat.LOWER_UNDERSCORE
+) : JsonAdapter<Any>() {
+    @Suppress("UNCHECKED_CAST")
+    private val builderType = try {
+        Class.forName("${messageType.name}\$Builder")
+    } catch (e: ClassNotFoundException) {
+        throw AssertionError("no builder for ${messageType.name}")
+    } as Class<Message.Builder<*, *>>
+
+    @Suppress("UNCHECKED_CAST")
+    private val newBuilder = builderType.constructors.find {
+        it.parameters.isEmpty()
+    } ?: throw AssertionError("no suitable constructor for ${builderType.name}")
+
+    private val fieldBindings: Map<String, FieldBinding> = messageType.declaredFields
+            .mapNotNull { field ->
+                field.getAnnotation(WireField::class.java)
+                        ?.let { FieldBinding(it, builderType, field, moshi) }
+            }.map { it.name to it }
+            .toMap()
+
+    override fun fromJson(reader: JsonReader): Any? {
+        val builder = newBuilder.newInstance() as Message.Builder<*, *>
+        reader.beginObject()
+        while (reader.hasNext()) {
+            val jsonPropertyName = reader.nextName()
+            val fieldName = jsonCaseFormat.to(CaseFormat.LOWER_UNDERSCORE, jsonPropertyName)
```

Comment:
Yikes. Can we skip case mapping? I'm not a fan.

https://publicobject.com/2016/01/20/strict-naming-conventions-are-a-liability/

Reply:
lol. sure, removed